Makefile

```diff
@@ -0,0 +1,30 @@
+CC := gcc
+
+LIBS := -lstdc++ -lavcodec -lavformat -lavutil -lswresample -lswscale -lglfw -lGL
+TARGET_DIR := bin/
+
+SRC_FILES_PLAYER := main_window.cpp decoder.cpp
+SRC_FILES_TCP_PLAYER := tcp_main_window.cpp decoder.cpp tcp_client.c
+SRC_FILES_TCP_SERVER := tcp_server.c
+SRC_FILES_TRANSCODER := transcoder.c debug.c
+
+.PHONY: all
+
+all: dir player tcp_player tcp_server transcoder
+
+dir:
+	if [ ! -d "$(TARGET_DIR)" ]; then \
+		mkdir -p $(TARGET_DIR); \
+	fi
+	
+player:
+	$(CC) $(SRC_FILES_PLAYER) $(LIBS) -o $(TARGET_DIR)$@
```

Comment:
Can you check if -o itself is the way to mention in the makefile that the output binary should land in the target dir?